npm/test/auth.test.ts

```diff
@@ -0,0 +1,46 @@
+import { getToken } from "../src/auth"
+jest.mock("keycloak-js")
+
+import Keycloak from "keycloak-js"
+
+class MockKeycloakAuthenticated {
+  token: string = "fake-auth-token"
+
+  init() {
+    return new Promise((resolve, _) => resolve(true))
+  }
+}
+class MockKeycloakUnauthenticated {
+  init() {
+    return new Promise((resolve, _) => resolve(false))
+  }
+}
+class MockKeycloakError {
+  init() {
+    return new Promise((_, reject) => reject("There has been an error"))
+  }
+}
+
+beforeEach(() => jest.resetModules())
+
+test("Returns an error if the user is not logged in", async () => {
+  ;(Keycloak as jest.Mock).mockImplementation(() => {
```

Comment:
Typescript novice question: what does the semicolon at the start of the line do?

Reply:
This is a prettier thing rather than specifically typescript. This is the only rationale I could find for it https://prettier.io/docs/en/rationale.html#semicolons which is a bit vague.

Reply:
Yeah, that sounds like the reason. It still looks ugly to me, but it makes sense. Ah, JavaScript.

Reply:
It's a bit weird, I agree. One of the main reasons for using something like prettier is that it takes personal preference out of the formatting. I don't think there's any configuration that can change this odd semcolon thing but if anyone wants to change any of the prettier config to something else, longer lines or whatever, then that's fine. So long as its consistent :smile: